pkg/api/types.go

```diff
@@ -1077,8 +1077,14 @@ type Event struct {
 	// Optional. The component reporting this event. Should be a short machine understandable string.
 	Source EventSource `json:"source,omitempty"`
 
-	// The time at which the client recorded the event. (Time of server receipt is in TypeMeta.)
-	Timestamp util.Time `json:"timestamp,omitempty"`
+	// The time at which the event was first recorded. (Time of server receipt is in TypeMeta.)
+	InitialTimestamp util.Time `json:"initialTimestamp,omitempty"`
+
+	// The time at which the most recent occurance of this event was recorded.
+	LastSeenTimestamp util.Time `json:"initialTimestamp,omitempty"`
+
+	// The number of times this event has occurred.
```

Comment:
Can you talk a bit about semantics?  Is this in perpetuity?  Is there a time horizon beyond which you store a new first event?  How exact a match does an event have to be to be considered a duplicate?  I assume that timestamp and count are ignored when coming from a user?